src/azul/indexer/document_service.py

```diff
@@ -68,6 +72,8 @@ def field_type(self, catalog: CatalogName, path: Tuple[str, ...]) -> FieldType:
                 raise KeyError(f'Path {path} not represented in field_types')
             except TypeError:
                 raise TypeError(f'Path {path} not represented in field_types')
+            if isinstance(field_types, list):
```

Comment:
This would handle inner nodes that are lists, which we currently don't support. I think this function is broken in other ways, like it doesnt handle partial paths, or the return type is inaccurate, but you won't need to fix that here.

Reply:
I did that intentionally, but I see why it would make sense to wait until that feature is actually implemented.

Reply:
You make it sound as if there is a trade-off here. If it was your intention to handle a case that shouldn't actually occur, that intention was misguided. Your implementation is unsound (its parts are inconsistent) and therefore confusing at best, and incorrect at worst.

Reply:
All I was trying to do in my last comment was admit my mistake / misunderstanding. Any other interpretations were unintended.